File: backend/app/models/http/schemas.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.http.enums import DocumentStatus, TicketSource, TicketStatus, UserRole
from app.models.internal.domain import TopQuestionRow


class DocumentResponse(BaseModel):
    id: uuid.UUID
    filename: str
    status: DocumentStatus
    mime_type: str | None = None
    file_size_bytes: int | None = None
    page_count: int | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class IngestResponse(BaseModel):
    document_id: uuid.UUID
    status: DocumentStatus = "pending"
    message: str = "Document queued for processing"


class Citation(BaseModel):
    chunk_id: uuid.UUID | None = None
    document_id: uuid.UUID | None = None
    document_name: str
    page_number: int | None = None
    chunk_text: str
    excerpt: str | None = None
    exact_quote_highlight: str | None = None
    citation_index: int | None = None


class QueryRequest(BaseModel):
    query: str = Field(..., min_length=1)
    session_id: str | None = Field(default=None, max_length=64)


class QueryResponse(BaseModel):
    answer: str
    citations: list[Citation] = []
    latency_ms: int | None = None


class UserProfileResponse(BaseModel):
    id: uuid.UUID
    email: str
    full_name: str | None = None
    job_title: str | None = None
    role: UserRole
    org_id: uuid.UUID | None = None
    department: str | None = None
    avatar_url: str | None = None

    model_config = {"from_attributes": True}


class TicketResponse(BaseModel):
    id: uuid.UUID
    source: TicketSource
    title: str | None = None
    intent: str | None
    priority: int | None
    summary: str | None
    department: str | None
    status: TicketStatus
    assignee_email: str | None = None
    assignee_id: uuid.UUID | None = None
    assignee_name: str | None = None
    slack_channel_id: str | None = None
    due_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class AnalyticsDashboard(BaseModel):
    queries_today: int
    latency_p50_ms: int | None
    latency_p95_ms: int | None
    documents_indexed: int
    top_questions: list[TopQuestionRow]

File: frontend/src/common/types/http/schemas.ts
/** Mirrors backend `app/models/http/schemas.py` (JSON field shapes). */

import type {
  DocumentStatus,
  TicketSource,
  TicketStatus,
  UserRole,
} from "@/common/types/http/enums";

export type DocumentRecord = {
  id: string;
  filename: string;
  status: DocumentStatus;
  mime_type?: string | null;
  file_size_bytes?: number | null;
  page_count?: number | null;
  created_at: string;
  updated_at: string;
};

export type IngestResponse = {
  document_id: string;
  status: DocumentStatus;
  message: string;
};

export type Citation = {
  chunk_id?: string | null;
  document_id?: string | null;
  document_name: string;
  page_number: number | null;
  chunk_text: string;
  excerpt?: string | null;
  exact_quote_highlight?: string | null;
  citation_index?: number | null;
};

export type QueryRequest = {
  query: string;
  session_id?: string | null;
};

export type QueryResponse = {
  answer: string;
  citations: Citation[];
  latency_ms: number | null;
};

/** Alias used by chat hooks and slice. */
export type QueryResult = QueryResponse;

export type TopQuestionRow = {
  question: string;
  count: number;
};

export type AnalyticsDashboard = {
  queries_today: number;
  latency_p50_ms: number | null;
  latency_p95_ms: number | null;
  documents_indexed: number;
  top_questions: TopQuestionRow[];
};

export type TicketRecord = {
  id: string;
  source: TicketSource;
  title?: string | null;
  intent: string | null;
  priority: number | null;
  summary: string | null;
  department: string | null;
  status: TicketStatus;
  assignee_email: string | null;
  assignee_id: string | null;
  assignee_name?: string | null;
  slack_channel_id: string | null;
  due_at?: string | null;
  created_at: string;
};

export type UserProfile = {
  id: string;
  email: string;
  full_name?: string | null;
  job_title?: string | null;
  role: UserRole;
  org_id?: string | null;
  department?: string | null;
  avatar_url?: string | null;
};
